workflow/common/util.go

```diff
@@ -454,23 +454,19 @@ func GetTaskAncestry(taskName string, tasks []wfv1.DAGTask) []string {
 		taskByName[task.Name] = task
 	}
 
-	visited := make(map[string]bool)
+	ancestry := make([]string, 0)
```

Comment:
The reason this was a map instead of list, because with a list, ancestors could appear twice in the ancestry (e.g. with a  diamond DAG). We need to dedupe the returned list, but we just need to do it in a deterministic manner.

I propose that we fix the bug this by sorting the ancestry list before returning.

Reply:
I also thought of this problem but then we need careful with the sorting as we should sort the ancestors by their finished time (their corresponding node status `finishedAt`)?

So that the last finished task is the last element and then its artifacts are used as global artifacts.
Otherwise global artifacts of parallel steps are determined just by their order in the `dependencies` array.
For example when we have a workflow like the diamond DAG where B and C have the same global artifacts:
```
   A
  / \
 B   C
  \ /
   D
```
In this case everything D gets the artifacts of C just because of its order.
If B finishes after C I think D should get the artifacts from B.

@jessesuen whats your opinion on that?

Reply:
I think sorting by finishedTime would work better. Great suggestion.

Reply:
However, not all nodes have a finishedAt if they are still running. Need to make sure, when finishedAt is not set, that returned order is still deterministic. Maybe fall back to startedAt.

Reply:
I adjusted my changes and and also fall back to the `startedAt` time if the `finishedAt` is Zero.
As the function now depends on the `dagContext` to get the node's 
status, I had to add a interface type and make the `GetTaskNode` function of the `dagContext` public to be accessible within the common package.
As this function is a dependency to the validation process, the current time is returned which returns the list in a deterministic order which is sufficient for the validation part.